test/src/Model/BrushFaceTest.cpp

```diff
@@ -106,5 +113,269 @@ namespace TrenchBroom {
+            for (int i=0; i<(1 << 3); i++) {
+                Mat4x4 xform;
+                
+                const bool testRoll    = (i & (1 << 0)) != 0;
+                const bool testPitch   = (i & (1 << 1)) != 0;
+                const bool testYaw     = (i & (1 << 2)) != 0;
+                
+                if (testRoll) {
+                    xform = rotationMatrix(rotateRadians, 0.0, 0.0) * xform;
+                }
+                if (testPitch) {
+                    xform = rotationMatrix(0.0, rotateRadians, 0.0) * xform;
+                }
+                if (testYaw) {
+                    xform = rotationMatrix(0.0, 0.0, rotateRadians) * xform;
+                }
+
+                checkTextureLockOnWithTransform(xform, origFace);
+            }
+        }
+        
+        /**
+         * Tests texture lock by rotating +/- the given amount, in one axis at a time.
+         */
+        static void checkTextureLockWithSingleAxisRotations(const BrushFace *origFace,
+                                                            double degrees) {
+            const double rotateRadians = Math::radians(degrees);
+            
+            for (int i=0; i<6; i++) {
+                Mat4x4 xform;
+                
+                switch (i) {
+                    case 0: xform = rotationMatrix(rotateRadians, 0.0, 0.0) * xform; break;
+                    case 1: xform = rotationMatrix(-rotateRadians, 0.0, 0.0) * xform; break;
+                    case 2: xform = rotationMatrix(0.0, rotateRadians, 0.0) * xform; break;
+                    case 3: xform = rotationMatrix(0.0, -rotateRadians, 0.0) * xform; break;
+                    case 4: xform = rotationMatrix(0.0, 0.0, rotateRadians) * xform; break;
+                    case 5: xform = rotationMatrix(0.0, 0.0, -rotateRadians) * xform; break;
+                }
+                
+                checkTextureLockOnWithTransform(xform, origFace);
+            }
+        }
+        
+        static void checkTextureLockOffWithTranslation(const BrushFace *origFace) {
+            Mat4x4 xform = translationMatrix(Vec3(100.0, 100.0, 100.0));
+            checkTextureLockOffWithTransform(xform, origFace);
+        }
+        
+        static void checkTextureLockForFace(const BrushFace *origFace, bool doParallelTests) {
+            checkTextureLockWithTranslationAnd90DegreeRotations(origFace);
+            checkTextureLockWithSingleAxisRotations(origFace, 30);
+            checkTextureLockWithSingleAxisRotations(origFace, 45);
+            
+            // rotation on multiple axes simultaneously is only expected to work on ParallelTexCoordSystem
+            if (doParallelTests) {
+                checkTextureLockWithMultiAxisRotations(origFace, 30);
+                checkTextureLockWithMultiAxisRotations(origFace, 45);
+            }
+            
+            checkTextureLockOffWithTranslation(origFace);
+        }
+        
+        TEST(BrushFaceTest, testTextureLock_Paraxial) {
```

Comment:
Thanks for the test cases, looks like it was more work than the actual changes ;-)